r/tests/testthat/test-compute-aggregate.R

```diff
@@ -383,33 +383,30 @@ test_that("value_counts", {
 })
 
 test_that("any.Array and any.ChunkedArray", {
-  
   data <- c(1:10, NA, NA)
 
-  expect_vector_equal(any(input > 5), data)
+  expect_vector_equal(any(input > 5, na.rm = TRUE), data)
```

Comment:
Does this test fail without the `, na.rm = TRUE`?

Reply:
It does indeed:
```R
Failure (test-compute-aggregate.R:388:3): any.Array and any.ChunkedArray
as.vector(x) not equal to `y`.
'is.NA' value mismatch: 0 in current 1 in target
Backtrace:
 1. expect_vector_equal(any(input > 5), data) test-compute-aggregate.R:388:2
 2. expect_as_vector(via_array, expected, ignore_attr, ...) helper-expectation.R:174:4
 3. testthat:::expect_fun(as.vector(x), y, ...) helper-expectation.R:24:2

Failure (test-compute-aggregate.R:388:3): any.Array and any.ChunkedArray
as.vector(x) not equal to `y`.
'is.NA' value mismatch: 0 in current 1 in target
Backtrace:
 1. expect_vector_equal(any(input > 5), data) test-compute-aggregate.R:388:2
 2. expect_as_vector(via_chunked, expected, ignore_attr, ...) helper-expectation.R:187:4
 3. testthat:::expect_fun(as.vector(x), y, ...) helper-expectation.R:24:2
```

Reply:
I'm not sure how to read what happens in `expect_vector_equal`.

Reply:
[`expect_vector_equal` takes an expression and a vector](https://github.com/apache/arrow/blob/ed4f79c77e5d34d5996fa287df0934dd2f93a12a/r/tests/testthat/helper-expectation.R#L159-L195) and evaluates the expression in base R, then converts the vector converts it into an array and evaluates the expression again (which typically will call arrow compute functions), and finally converts the vector to a chunked array and does the same.

Reply:
So the errors you're seeing indicate that we've got a mismatch in behavior. I see that Ian has commented below about more robust tests + how to match the behavior (but I'm happy to add to it if that's unclear!)

Reply:
Indeed @jonkeane it was a behavior issue. Should be fixed now.